src/experimental/Db.h

```diff
@@ -51,112 +51,58 @@ class Entry {
 // constructor
 class PilotEntry : public Entry {
     PilotEntry() = delete;
+public:
     PilotEntry(int row_id) : Entry::Entry(DataPosition("pilots", row_id)) {}
-};
-
-Entry newPilotEntry(int row_id) { return Entry(DataPosition("pilots", row_id)); }
-
-static
-bool insertPilot(UserInput& uin)
-{
-    DEB("Inserting...");
-    auto data = uin.all();
-    auto position = DEFAULT_PILOT_POSITION;
-
-    if (data.isEmpty()) {
-        DEB("Object Contains no data. Aborting.");
-        return false;
-    }
-    QString statement = "INSERT INTO " + position.first + QLatin1String(" (");
-    for (auto i = data.cbegin(); i != data.cend(); ++i) {
-        statement += i.key() + QLatin1String(", ");
+    PilotEntry(TableData fromNewPilotDialog) : Entry::Entry(DataPosition("pilots", 0)) {
+        setData(fromNewPilotDialog);
     }
-    statement.chop(2);
-    statement += QLatin1String(") VALUES (");
-    for (auto i = data.cbegin(); i != data.cend(); ++i) {
-        statement += QLatin1String("'") + i.value() + QLatin1String("', ");
-    }
-    statement.chop(2);
-    statement += QLatin1String(")");
-
-    QSqlQuery q(statement);
-    if (q.lastError().type() == QSqlError::NoError) {
-        DEB("Entry successfully committed.");
-        return true;
-    } else {
-        DEB("Unable to commit. Query Error: " << q.lastError().text());
-        return false;
-    }
-}
+};
 
 /*!
  * \brief The DB class encapsulates the SQL database by providing fast access
  * to hot database data.
  */
 class DB {
 private:
-    static QStringList tableNames;
-    static QMap<TableName, QStringList> tableColumns;
+    static inline QStringList                   tableNames;
+    static inline QMap<TableName, QStringList>  tableColumns;
```

Comment:
what does `inline` provide here?

Reply:
Wouldn't compile without it, the problem ocurred when connect() tried writing to these variables. I found [this](url=https://stackoverflow.com/questions/9656941/why-cant-i-initialize-non-const-static-member-or-static-array-in-class) explication and went with it.



> It's because there can only be one definition of A::a that all the translation units use.

>If you performed static int a = 3; in a class in a header included in all a translation units then you'd get multiple definitions. Therefore, non out-of-line definition of a static is forcibly made a compiler error.

>Using static inline or static const remedies this. static inline only concretises the symbol if it is used in the translation unit and ensures the linker only selects and leaves one copy if it's defined in multiple translation units due to it being in a comdat group. const at file scope makes the compiler never emit a symbol because it's always substituted immediately in the code unless extern is used, which is not permitted in a class.